pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -90,3 +97,59 @@ func (m *Manager) updateEnactmentConditions(
 		return nil
 	})
 }
+
+func (m *Manager) refreshPolicyConditions() error {
+
+	// On conflict we need to re-retrieve enactments since the
+	// conflict can denote that the calculated policy conditions
+	// are now not accurate.
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		enactments := nmstatev1alpha1.NodeNetworkConfigurationEnactmentList{}
```

Comment:
Let's stop here for a second. We have retry on conflict, so the overall conditions we read are for sure latest. Do we need to fetch all the enactments here or can we do the calculation based only on what we have in the overall conditions and our conditions?

I'm too tired to tell for sure, but I feel there may be a way to do the calculation without fetching all the Enactments. Some fun programming finally!

Reply:
This would be kind of the algorithm in case we don't wan to count always:

NotifyFailure:
- If overall is failure we change the counter
- If overall is progressing we notify failure
- if it's success we have do something wrong
NotifyProgress:
- if overall is failure we do nothing
- if overall is progressing we change the counter.
- if it's success we have do something wrong
NotifySuccess:
- We have to look at the whole cluster and change to success is everything is allright.
NotifyNotMatching:
- if overall is failure we do nothing
- if overall is progressing we decrease the counter 
- if overall is success we did something wrong
- else we we have to look at the whole cluster and set sucess to false.

I see this more complex than a refresh function at controller (selecting it with leader election) with a refresh timer.